File: non_native/lib/view/components/bird_card.dart
import 'package:flutter/material.dart';

import '../../util/colors.dart';
import 'custom_button.dart';
import 'custom_white_text.dart';

class BirdCard extends StatefulWidget {
  final String name;
  final String order;
  final String family;
  final String habitatTemp;
  final int number;
  final VoidCallback onEditClick;
  final VoidCallback onDeleteClick;

  const BirdCard({
    super.key,
    required this.name,
    required this.order,
    required this.family,
    required this.habitatTemp,
    required this.number,
    required this.onEditClick,
    required this.onDeleteClick,
  });

  @override
  State<BirdCard> createState() => _BirdCardState();
}

class _BirdCardState extends State<BirdCard> {
  bool enabledButtons = true;

  @override
  Widget build(BuildContext context) {
    Color myColor = AppColors.darkGreen;
    Color myEditColor = AppColors.lightGreen;
    Color myDeleteColor = AppColors.darkOrange;

    return Card(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(20),
      ),
      color: myColor,
      child: Padding(
        padding: const EdgeInsets.all(20.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            CustomWhiteText(text: widget.name, size: 24),
            const SizedBox(
              height: 12,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                CustomWhiteText(text: widget.order),
                const SizedBox(width: 5),
                CustomWhiteText(text: widget.family),
              ],
            ),
            const SizedBox(
              height: 12,
            ),
            CustomWhiteText(text: widget.habitatTemp.toUpperCase()),
            const SizedBox(
              height: 12,
            ),
            CustomWhiteText(text: "Seen ${widget.number} time(s)", size: 12),
            const SizedBox(
              height: 12,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Expanded(
                  child: CustomButton(
                    text: "EDIT",
                    backGroundColor: myEditColor,
                    enabled: enabledButtons,
                    onClick: () {
                      setState(() {
                        enabledButtons = false;
                      });
                      widget.onEditClick();
                    },
                  ),
                ),
                const SizedBox(width: 5),
                Expanded(
                  child: CustomButton(
                    text: "DELETE",
                    backGroundColor: myDeleteColor,
                    enabled: enabledButtons,
                    onClick: widget.onDeleteClick,
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}

File: native_upgrade/app/src/main/java/com/example/myapplication/ui/components/BirdCard.kt
package com.example.myapplication.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.myapplication.ui.theme.DarkGreen
import com.example.myapplication.ui.theme.DarkOrange
import com.example.myapplication.ui.theme.LightGreen


@Composable
fun BirdCard(
    name: String,
    order: String,
    family: String,
    habitatTemp: String,
    number: Int,
    onEditClick: () -> Unit,
    onDeleteClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val myColor = DarkGreen
    val myEditColor = LightGreen
    val myDeleteColor = DarkOrange

    var enabledButtons by remember {
        mutableStateOf(true)
    }

    Column(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(myColor)
            .padding(start = 15.dp, end = 15.dp, top = 20.dp, bottom = 15.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        CustomWhiteText(text = name, size = 24.sp)
        Row {
            CustomWhiteText(text = order)
            Spacer(modifier = Modifier.width(5.dp))
            CustomWhiteText(text = family)
        }
        CustomWhiteText(text = habitatTemp)
        CustomWhiteText(text = "Seen $number time(s)", size = 12.sp)
        Row {
            CustomButton(
                text = "EDIT",
                backGroundColor = myEditColor,
                modifier.weight(1f),
                enabled = enabledButtons
            ) {
                enabledButtons = false
                onEditClick()
            }
            Spacer(modifier = Modifier.width(5.dp))
            CustomButton(
                text = "DELETE",
                backGroundColor = myDeleteColor,
                modifier.weight(1f),
                enabled = enabledButtons
            ) {
                onDeleteClick()
            }
        }
    }
}